fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1440,10 +1454,15 @@ void ReadYourWritesTransaction::writeRangeToNativeTransaction( KeyRangeRef const
 				switch(op[i].type) {
 					case MutationRef::SetValue:
 						if (op[i].value.present()) {
-							tr.set( it.beginKey().assertRef(), op[i].value.get(), false );
-						} else {
-							tr.clear( it.beginKey().assertRef(), false );
-						}
+						    approximateSize += updateSize ? it.beginKey().assertRef().expectedSize() +
+						                                        op[i].value.get().expectedSize() + sizeof(MutationRef)
+						                                  : 0;
+						    tr.set(it.beginKey().assertRef(), op[i].value.get(), false);
+					    } else {
+						    approximateSize +=
+						        updateSize ? it.beginKey().assertRef().expectedSize() + sizeof(MutationRef) : 0;
+						    tr.clear(it.beginKey().assertRef(), false);
```

Comment:
I think a single key clear will be stored as a clear range mutation, so you'll have to do the `2 * size + 1` logic here too.

Reply:
Also probably irrelevant